src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +36,12 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .mapToDouble(i -> numbers.get(i) - i % 2)
+                .filter(n -> n % 2 != 0)
+                .average()
+                .orElseThrow(() -> new NoSuchElementException(
+                        "Can't get average from list: " + numbers));
```

Comment:
If you need to throw exception when no result found after stream execution sometimes get()/getAsDouble() may help, try to google what does it do
I do same mistake and after read try to avoid) 
```suggestion
                .getAsDouble();
```